h/services/annotation_moderation.py

```diff
@@ -1,74 +1,26 @@
-from h import models
+from h.models import AnnotationModeration
 
 
 class AnnotationModerationService:
     def __init__(self, session):
-        self.session = session
-
-    @staticmethod
-    def hidden(annotation):
-        """
-        Check if an annotation id is hidden.
-
-        :param annotation: The annotation to check.
-        :type annotation: h.models.Annotation
-
-        :returns: true/false boolean
-        :rtype: bool
-        """
-        return annotation.moderation is not None
+        self._session = session
 
     def all_hidden(self, annotation_ids):
         """
         Check which of the given annotation ids is hidden.
 
         :param annotation_ids: The ids of the annotations to check.
-        :type annotation_ids: list of unicode
-
         :returns: The subset of the annotation ids that are hidden.
-        :rtype: set of unicode
         """
         if not annotation_ids:
             return set()
 
-        query = self.session.query(models.AnnotationModeration.annotation_id).filter(
-            models.AnnotationModeration.annotation_id.in_(annotation_ids)
+        query = self._session.query(AnnotationModeration.annotation_id).filter(
+            AnnotationModeration.annotation_id.in_(annotation_ids)
```

Comment:
Not sure if I understand AnnotationModeration, I'm probably missing some extra functionality, but looks like currently it could be replaced by something like a null-able `flagged_at` on Annotation.

Reply:
I'll be 100% honest and say I haven't looked into it. But looking at it... _yeah_? The only thing that table has is a back-reference to the annotation. Could totally be a boolean.

I can see it making sense if the moderation had some extra details like who moderated it and when.

I've also been trying to steer clear of things that involve DB migrations as it scares me.